client/src/components/Auth/CheckLogged.js

```diff
@@ -0,0 +1,33 @@
+import React from "react";
```

Comment:
CheckLogged and WithAuth are very similar. Do you need both?
The usage of the second argument `admin` is confusing, you could pass the path to redirect to on success directly.